lib/util/news.ts

```diff
@@ -0,0 +1,43 @@
+import type { Languages } from '../hooks/useLanguage'
+import { languages } from '../hooks/useLanguage'
+import { z } from 'zod'
+
+export type News = {
+  title: string,
+  date: Date,
+  description: (string | URL)[],
+  externalResource?: URL,
+  keys: string[]
+}
+
+export type LocalizedNews = Record<Languages, News[]>
+
+export const newsSchema = z.object({
+  title: z.string(),
+  description: z.string(),
+  date: z.string(),
+  image: z.string().url().optional(),
+  externalResource: z.string().url().optional(),
+  keys: z.array(z.string())
+}).transform<News>((obj) => {
+  let description: (string | URL)[] = [obj.description]
+  if (obj.image) {
+    description = [new URL(obj.image), ...description]
+  }
+
+  return {
+    title: obj.title,
+    date: new Date(obj.date),
```

Comment:
what the date constructor can parse may differ between browsers. The spec doesn't state all too much in this regard (afaik) and many browsers just started to do some kind of "best effort" strategy and started supporting many many different formats, thus this is kind of a mess and sticking to ISO 8601 would definitely be the most safe way forward